http://www.phorum.org/phorum5/read.php?61,131083

--- phorum/htdocs/read.php~	2010-05-03 18:12:21.000000000 +0300
+++ phorum/htdocs/read.php	2010-05-04 11:38:50.829674278 +0300
@@ -788,7 +788,18 @@
     // here, because of modules adding images and formatting.
     $PHORUM["DATA"]["HTML_TITLE"] = trim(strip_tags($PHORUM["threaded_read"] ? $PHORUM["DATA"]["MESSAGE"]["subject"] : $PHORUM["DATA"]["TOPIC"]["subject"]));
 
-    $PHORUM["DATA"]["DESCRIPTION"] = htmlspecialchars(preg_replace('!\s+!s'," ",strip_tags(substr($PHORUM["DATA"]["TOPIC"]["body"],0,300))), ENT_COMPAT, $PHORUM["DATA"]["HCHARSET"]);
+    $PHORUM["DATA"]["DESCRIPTION"] = preg_replace('!\s+!s'," ", strip_tags($PHORUM["DATA"]["TOPIC"]["body"]));
+    if (strtoupper($PHORUM["DATA"]["HCHARSET"])=='UTF-8') {
+        if (function_exists('mb_substr')) {
+            $PHORUM["DATA"]["DESCRIPTION"] = mb_substr($PHORUM["DATA"]["DESCRIPTION"], 0, 300, $PHORUM["DATA"]["HCHARSET"]);
+        } else {
+            // No change to short description without PHP warning
+            $PHORUM["DATA"]["DESCRIPTION"] = $PHORUM["DATA"]["DESCRIPTION"];
+        }
+    } else {
+        $PHORUM["DATA"]["DESCRIPTION"] = substr($PHORUM["DATA"]["DESCRIPTION"], 0, 300);
+    }
+    $PHORUM["DATA"]["DESCRIPTION"] = htmlspecialchars($PHORUM["DATA"]["DESCRIPTION"], ENT_COMPAT, $PHORUM["DATA"]["HCHARSET"]);
 
     // add feed url
     if(isset($PHORUM['use_rss']) && $PHORUM['use_rss']){
