--- phorum-5.2.14/include/db/config.php.sample~	2010-01-06 19:24:41.000000000 +0200
+++ phorum-5.2.14/include/db/config.php.sample	2010-01-06 19:24:57.091519006 +0200
@@ -17,8 +17,8 @@
     'type'          => 'mysql',
     'name'          => 'phorum5',
     'server'        => 'localhost',
-    'user'          => 'phorum5',
-    'password'      => 'phorum5',
+    'user'          => 'mysql',
+    'password'      => '',
     'table_prefix'  => 'phorum',
     'port'          => '3306',
     'socket'        => NULL,
