--- phorum-5.1.25/common.php~	2007-12-27 18:43:19.000000000 +0200
+++ phorum-5.1.25/common.php	2007-12-27 18:46:34.201598153 +0200
@@ -107,17 +107,17 @@
     @ini_set("display_errors", 0);
 
     // Load configuration.
-    if (! include_once( "./include/db/config.php" )) {
+    if (! include_once( "/etc/webapps/phorum/config.php" )) {
         print '<html><head><title>Phorum error</title></head><body>';
         print '<h2>Phorum database configuration error</h2>';
 
         // No database configuration found.
-        if (!file_exists("./include/db/config.php")) { ?>
+        if (!file_exists("/etc/webapps/phorum/config.php")) { ?>
             Phorum has been installed on this server, but the configuration<br/>
             for the database connection has not yet been made. Please read<br/>
             <a href="docs/install.txt">docs/install.txt</a> for installation instructions. <?php
         } else {
-            $fp = fopen("./include/db/config.php", "r");
+            $fp = fopen("/etc/webapps/phorum/config.php", "r");
             // Unable to read the configuration file.
             if (!$fp) { ?>
                 A database configuration file was found in ./include/db/config.php,<br/>
@@ -151,7 +151,7 @@
         header("Location: ".$PHORUM["DBCONFIG"]["down_page"]);
         exit();
     } else {
-        echo "The database connection failed. Please check your database configuration in include/db/config.php. If the configuration is okay, check if the database server is running.";
+        echo "The database connection failed. Please check your database configuration in /etc/webapps/phorum/config.php. If the configuration is okay, check if the database server is running.";
         exit();
     }
 }
--- ./include/db/mysql.php~	2005-10-21 23:27:53.000000000 +0300
+++ ./include/db/mysql.php	2006-03-08 17:56:25.213522731 +0200
@@ -10,7 +10,7 @@
  *    few tables as it likes.  It can store the fields anyway it wants.
  *    The only thing to worry about is the table_prefix for the tables.
  *    all tables for a Phorum install should be prefixed with the
- *    table_prefix that will be entered in include/db/config.php.  This
+ *    table_prefix that will be entered in /etc/webapps/phorum/config.php.  This
  *    will allow multiple Phorum installations to use the same database.
  */
 
--- ./include/admin/install.php~	2005-03-17 19:41:22.000000000 +0200
+++ ./include/admin/install.php	2006-03-08 17:56:43.134679959 +0200
@@ -3,7 +3,7 @@
     if(!defined("PHORUM_ADMIN")) return;
 
     if(!phorum_db_check_connection()){
-        echo "A database connection could not be established.  Please edit include/db/config.php.";
+        echo "A database connection could not be established.  Please edit /etc/webapps/phorum/config.php.";
         return;
     }
 
--- phorum-5.0.21/admin.php	2005-03-17 19:41:22.000000000 +0200
+++ phorum-5.0.21.pld/admin.php	2006-03-08 18:23:04.000000000 +0200
@@ -26,6 +26,7 @@
     // the code run faster.
     error_reporting  (E_ERROR | E_WARNING | E_PARSE);
 
+    chdir("..");
     include_once "./common.php";
     include_once "./include/users.php";
 
--- phorum-5.0.21/control.php	2005-10-29 00:02:51.000000000 +0300
+++ phorum-5.0.21.pld/control.php	2006-03-08 18:23:04.000000000 +0200
@@ -18,6 +18,7 @@
 ////////////////////////////////////////////////////////////////////////////////
 define('phorum_page','control');
 
+chdir("..");
 include_once("./common.php");
 
 phorum_require_login();
--- phorum-5.0.21/file.php	2005-03-17 19:41:22.000000000 +0200
+++ phorum-5.0.21.pld/file.php	2006-03-08 18:23:04.000000000 +0200
@@ -23,6 +23,7 @@
 ini_set ( "zlib.output_compression", "0");
 ini_set ( "output_handler", "");
 
+chdir("..");
 include_once("./common.php");
 
 // checking read-permissions
--- phorum-5.0.21/follow.php	2005-03-17 19:41:22.000000000 +0200
+++ phorum-5.0.21.pld/follow.php	2006-03-08 18:23:04.000000000 +0200
@@ -18,6 +18,7 @@
 ////////////////////////////////////////////////////////////////////////////////
 define('phorum_page','subscribe');
 
+chdir("..");
 include_once("./common.php");
 
 phorum_require_login();
--- phorum-5.0.21/index.php	2005-03-17 19:41:22.000000000 +0200
+++ phorum-5.0.21.pld/index.php	2006-03-08 18:23:04.000000000 +0200
@@ -17,6 +17,7 @@
 ///////////////////////////////////////////////////////////////////////////////
 define('phorum_page','index');
 
+chdir("..");
 include_once( "./common.php" );
 
 include_once( "./include/format_functions.php" );
--- phorum-5.0.21/list.php	2005-04-01 20:10:41.000000000 +0300
+++ phorum-5.0.21.pld/list.php	2006-03-08 18:23:04.000000000 +0200
@@ -18,6 +18,7 @@
 ////////////////////////////////////////////////////////////////////////////////
 define('phorum_page','list');
 
+chdir("..");
 include_once("./common.php");
 include_once("./include/format_functions.php");
 
--- phorum-5.0.21/login.php	2005-10-15 13:51:01.000000000 +0300
+++ phorum-5.0.21.pld/login.php	2006-03-08 18:23:04.000000000 +0200
@@ -18,6 +18,7 @@
 ////////////////////////////////////////////////////////////////////////////////
 define('phorum_page','login');
 
+chdir("..");
 include_once( "./common.php" );
 include_once( "./include/users.php" );
 include_once( "./include/email_functions.php" );
--- phorum-5.0.21/moderation.php	2005-10-21 13:35:57.000000000 +0300
+++ phorum-5.0.21.pld/moderation.php	2006-03-08 18:23:04.000000000 +0200
@@ -18,6 +18,7 @@
 ////////////////////////////////////////////////////////////////////////////////
 define('phorum_page','moderation');
 
+chdir("..");
 include_once("./common.php");
 include_once("./include/moderation_functions.php");
 include_once("./include/thread_info.php");
--- phorum-5.1.25/posting.php~	2007-06-29 18:03:25.000000000 +0300
+++ phorum-5.1.25/posting.php	2007-12-27 18:49:04.318221622 +0200
@@ -55,6 +55,7 @@
     define('phorum_page', 'post');
 }
 
+chdir("..");
 include_once("./common.php");
 include_once("include/format_functions.php");
 
--- phorum-5.0.21/profile.php	2005-04-15 16:37:05.000000000 +0300
+++ phorum-5.0.21.pld/profile.php	2006-03-08 18:23:05.000000000 +0200
@@ -18,6 +18,7 @@
 ////////////////////////////////////////////////////////////////////////////////
 define('phorum_page','profile');
 
+chdir("..");
 include_once("./common.php");
 include_once("./include/email_functions.php");
 include_once("./include/format_functions.php");
--- phorum-5.0.21/read.php	2005-04-01 20:10:41.000000000 +0300
+++ phorum-5.0.21.pld/read.php	2006-03-08 18:23:05.000000000 +0200
@@ -18,6 +18,7 @@
 ////////////////////////////////////////////////////////////////////////////////
 define('phorum_page','read');
 
+chdir("..");
 include_once("./common.php");
 include_once("./include/email_functions.php");
 include_once("./include/format_functions.php");
--- phorum-5.0.21/register.php	2005-10-14 23:46:40.000000000 +0300
+++ phorum-5.0.21.pld/register.php	2006-03-08 18:23:05.000000000 +0200
@@ -18,6 +18,7 @@
 ////////////////////////////////////////////////////////////////////////////////
 define('phorum_page','register');
 
+chdir("..");
 include_once("./common.php");
 include_once("./include/users.php");
 include_once("./include/profile_functions.php");
--- phorum-5.0.21/script.php	2005-03-17 19:41:22.000000000 +0200
+++ phorum-5.0.21.pld/script.php	2006-03-08 18:23:05.000000000 +0200
@@ -18,6 +18,7 @@
 ////////////////////////////////////////////////////////////////////////////////
 define('phorum_page','script');
 
+chdir("..");
 include_once("./common.php");
 
 // if we are running in the webserver, bail out
--- phorum-5.0.21/search.php	2005-11-03 23:56:46.000000000 +0200
+++ phorum-5.0.21.pld/search.php	2006-03-08 18:23:05.000000000 +0200
@@ -18,6 +18,7 @@
 ////////////////////////////////////////////////////////////////////////////////
 define('phorum_page','search');
 
+chdir("..");
 include_once("./common.php");
 
 if(!phorum_check_read_common()) {
--- phorum-5.1.25/include/admin/install.php~	2007-12-27 18:47:40.000000000 +0200
+++ phorum-5.1.25/include/admin/install.php	2007-12-27 18:51:17.809890092 +0200
@@ -319,7 +319,7 @@
         case 5:
 
             phorum_db_update_settings( array("installed"=>1) );
-            echo "The setup is complete.  You can now go to <a href=\"$_SERVER[PHP_SELF]\">the admin</a> and start making Phorum all your own.<br /><br /><strong>Here are some things you will want to look at:</strong><br /><br /><a href=\"$_SERVER[PHP_SELF]?module=settings\">The General Settings page</a><br /><br /><a href=\"$_SERVER[PHP_SELF]?module=mods\">Pre-installed modules</a><br /><br /><a href=\"docs/faq.txt\">The FAQ</a><br /><br /><a href=\"docs/performance.txt\">How to get peak performance from Phorum</a><br /><br /><strong>For developers:</strong><br /><br /><a href=\"docs/creating_mods.txt\">Module Creation</a><br /><br /><a href=\"docs/permissions.txt\">How Phorum permisssions work</a><br /><br /><a href=\"docs/CODING-STANDARDS\">The Phorum Team's codings standards</a>";
+            echo "The setup is complete.  You can now go to <a href=\"$_SERVER[PHP_SELF]\">the admin</a> and start making Phorum all your own.<br /><br /><strong>Here are some things you will want to look at:</strong><br /><br /><a href=\"$_SERVER[PHP_SELF]?module=settings\">The General Settings page</a><br /><br /><a href=\"$_SERVER[PHP_SELF]?module=mods\">Pre-installed modules</a><br /><br /><a href=\"http://www.phorum.org/phorum5/docs/faq.txt\">The FAQ</a><br /><br /><a href=\"http://www.phorum.org/phorum5/docs/performance.txt\">How to get peak performance from Phorum</a><br /><br /><strong>For developers:</strong><br /><br /><a href=\"docs/creating_mods.txt\">Module Creation</a><br /><br /><a href=\"docs/permissions.txt\">How Phorum permisssions work</a><br /><br /><a href=\"docs/CODING-STANDARDS\">The Phorum Team's codings standards</a>";
 
             break;
 
