--- phorum/common.php	2010-01-12 19:23:19.129687819 +0200
+++ phorum/common.php	2010-01-12 19:36:17.512942870 +0200
@@ -20,6 +20,8 @@
 // Check that this file is not loaded directly.
 if ( basename( __FILE__ ) == basename( $_SERVER["PHP_SELF"] ) ) exit();
 
+define('PHORUM_DIR', dirname(__FILE__));
+define('PHORUM_INCLUDES_DIR', PHORUM_DIR.'/include');
 
 // ----------------------------------------------------------------------
 // Initialize variables and constants and load required libraries
@@ -82,22 +84,22 @@
     ob_start();
 
     // Load configuration.
-    if (! include_once( "./include/db/config.php" )) {
+    if (! include_once( "/etc/webapps/phorum/config.php" )) {
         print '<html><head><title>Phorum error</title></head><body>';
         print '<h2>Phorum database configuration error</h2>';
 
         // No database configuration found.
-        if (!file_exists("./include/db/config.php")) { ?>
+        if (!file_exists("/etc/webapps/phorum/config.php")) { ?>
             Phorum has been installed on this server, but the configuration<br/>
             for the database connection has not yet been made. Please read<br/>
             <a href="docs/install.txt">docs/install.txt</a> for installation
             instructions. <?php
         } else {
-            $fp = fopen("./include/db/config.php", "r");
+            $fp = fopen("/etc/webapps/phorum/config.php", "r");
             // Unable to read the configuration file.
             if (!$fp) { ?>
                 A database configuration file was found in
-                ./include/db/config.php,<br/>but Phorum was unable to read it.
+                config.php,<br/>but Phorum was unable to read it.
                 Please check the file permissions<br/>for this file. <?php
             // Unknown error.
             } else {
@@ -134,7 +136,7 @@
 
 // Load the database layer.
 $PHORUM['DBCONFIG']['type'] = basename($PHORUM['DBCONFIG']['type']);
-require_once( "./include/db/{$PHORUM['DBCONFIG']['type']}.php" );
+require_once PHORUM_INCLUDES_DIR."/db/{$PHORUM['DBCONFIG']['type']}.php";
 
 // Try to setup a connection to the database.
 if(!phorum_db_check_connection()){
@@ -217,7 +219,7 @@
 // Load the caching-layer. You can specify a different one in the settings.
 // One caching layer *needs* to be loaded.
 $PHORUM['cache_layer'] = basename($PHORUM['cache_layer']);
-require_once PHORUM_INCLUDES_DIR.'/cache/$PHORUM[cache_layer].php';
+require_once PHORUM_INCLUDES_DIR.'/cache/'.$PHORUM['cache_layer'].'.php';
 
 // Load phorum_get_url().
 // This function is used for generating all Phorum URLs.
@@ -227,7 +229,7 @@
 // module authors a chance to override them. This can be especially useful
 // for distibuting a module that contains a full Phorum template as well.
 // For switching, the function phorum_switch_template() can be used.
-$PHORUM['template_path'] = './templates';
+$PHORUM['template_path'] = PHORUM_DIR.'/templates';
 $PHORUM['template_http_path'] = $PHORUM['http_path'].'/templates';
 
 // ----------------------------------------------------------------------
@@ -691,9 +693,9 @@
 
     }
 
-    if ( !isset( $PHORUM["language"] ) || empty( $PHORUM["language"] ) || !file_exists( "./include/lang/$PHORUM[language].php" ) )
+    if (!isset($PHORUM["language"]) || empty($PHORUM["language"]) || !file_exists(PHORUM_INCLUDES_DIR."/lang/$PHORUM[language].php"))
         $PHORUM["language"] = $PHORUM["default_forum_options"]["language"];
-    if ( !file_exists("./include/lang/$PHORUM[language].php") ) {
+    if ( !file_exists(PHORUM_INCLUDES_DIR."/lang/{$PHORUM['language']}.php") ) {
         $PHORUM["language"] = PHORUM_DEFAULT_LANGUAGE;
     }
 
@@ -714,8 +716,8 @@
     }
 
     $PHORUM['language'] = basename($PHORUM['language']);
-    if ( file_exists( "./include/lang/$PHORUM[language].php" ) ) {
-        require_once PHORUM_INCLUDES_DIR.'/lang/$PHORUM[language].php';
+    if ( file_exists( PHORUM_INCLUDES_DIR."/lang/{$PHORUM['language']}.php" ) ) {
+        require_once PHORUM_INCLUDES_DIR."/lang/{$PHORUM['language']}.php";
     }
     // load languages for localized modules
     if ( isset( $PHORUM["hooks"]["lang"] ) && is_array($PHORUM["hooks"]["lang"]) ) {
@@ -723,11 +725,11 @@
         {
             // load mods for this hook
             $mod = basename($mod);
-            if ( file_exists( "./mods/$mod/lang/$PHORUM[language].php" ) ) {
-                require_once "./mods/$mod/lang/$PHORUM[language].php";
+            if ( file_exists(PHORUM_DIR."/mods/$mod/lang/$PHORUM[language].php" ) ) {
+                require_once PHORUM_DIR."/mods/$mod/lang/$PHORUM[language].php";
             }
-            elseif ( file_exists( "./mods/$mod/lang/".PHORUM_DEFAULT_LANGUAGE.".php" ) ) {
-                require_once "./mods/$mod/lang/".PHORUM_DEFAULT_LANGUAGE.".php";
+            elseif ( file_exists(PHORUM_DIR."/mods/$mod/lang/".PHORUM_DEFAULT_LANGUAGE.".php" ) ) {
+                require_once PHORUM_DIR."/mods/$mod/lang/".PHORUM_DEFAULT_LANGUAGE.".php";
             }
         }
     }
@@ -998,8 +1000,8 @@
     // strings at some point after all, for example if we reset the
     // author name in messages for deleted users to "anonymous".
     $PHORUM["language"] = basename($PHORUM["default_forum_options"]["language"]);
-    if (file_exists("./include/lang/$PHORUM[language].php")) {
-        require_once PHORUM_INCLUDES_DIR.'/lang/$PHORUM[language].php';
+    if (file_exists(PHORUM_INCLUDES_DIR."/lang/{$PHORUM['language']}.php")) {
+        require_once PHORUM_INCLUDES_DIR."/lang/{$PHORUM['language']}.php";
     }
 }
 
@@ -1308,7 +1310,7 @@
             // could be an incomplete or empty template.
             $postfix = '/info.php';
         } else {
-            $prefix = './mods/'.basename($module).'/templates';
+            $prefix = PHORUM_DIR.'/mods/'.basename($module).'/templates';
             $postfix = '';
         }
 
@@ -1669,10 +1671,10 @@
         foreach( $PHORUM["hooks"][$hook]["mods"] as $mod ) {
             // load mods for this hook
             $mod = basename($mod);
-            if ( file_exists( "./mods/$mod/$mod.php" ) ) {
-                require_once "./mods/$mod/$mod.php";
-            } elseif ( file_exists( "./mods/$mod.php" ) ) {
-                require_once "./mods/$mod.php";
+            if ( file_exists( PHORUM_DIR."/mods/$mod/$mod.php" ) ) {
+                require_once PHORUM_DIR."/mods/$mod/$mod.php";
+            } elseif ( file_exists( PHORUM_DIR."/mods/$mod.php" ) ) {
+                require_once PHORUM_DIR."/mods/$mod.php";
             }
         }
 
@@ -1759,10 +1761,10 @@
 
     $langs = array();
 
-    $d = dir( "./include/lang" );
+    $d = dir(PHORUM_INCLUDES_DIR."/lang" );
     while ( false !== ( $entry = $d->read() ) ) {
-        if ( substr( $entry, -4 ) == ".php" && is_file( "./include/lang/$entry" ) ) {
-            @include "./include/lang/$entry";
+        if ( substr( $entry, -4 ) == ".php" && is_file(PHORUM_INCLUDES_DIR."/lang/$entry" ) ) {
+            include PHORUM_INCLUDES_DIR."/lang/$entry";
             if ( !isset( $language_hide ) || empty( $language_hide ) || defined( "PHORUM_ADMIN" ) ) {
                 $langs[str_replace( ".php", "", $entry )] = $language;
             } else {
--- phorum-5.2.14/include/admin/install.php~	2010-01-06 19:23:20.000000000 +0200
+++ phorum-5.2.14/include/admin/install.php	2010-01-06 19:23:35.421527751 +0200
@@ -25,7 +25,7 @@
 include_once("./include/api/user.php");
 
 if(!phorum_db_check_connection()){
-    echo "A database connection could not be established.  Please edit include/db/config.php.";
+    echo "A database connection could not be established.  Please edit config.php.";
     return;
 }
 
--- phorum/include/admin/header.php~	2009-07-22 14:58:57.000000000 +0300
+++ phorum/include/admin/header.php	2010-01-06 20:02:05.938378394 +0200
@@ -36,7 +36,7 @@
 
     // set the path to the CSS file to pull in
     $default_admin_css_file = 'default.css';
-    $admin_css_path = $PHORUM["http_path"].'/include/admin/css/' . $default_admin_css_file;
+    $admin_css_path = $PHORUM["http_path"].'/admin/css/' . $default_admin_css_file;
 
     /**
      * [hook]
--- phorum/htdocs/admin/css/default.css~	2010-01-06 20:13:13.000000000 +0200
+++ phorum/htdocs/admin/css/default.css	2010-01-06 20:13:19.085201493 +0200
@@ -118,19 +118,19 @@
 .icon-folder-up {
     width: 22px;
     padding-left: 22px;
-    background: url(../../../images/folder_up.png) 0 1px no-repeat;
+    background: url(../../images/folder_up.png) 0 1px no-repeat;
 }
 
 .icon-folder {
     width: 22px;
     padding-left: 22px;
-    background: url(../../../images/folder.png) 0 1px no-repeat;
+    background: url(../../images/folder.png) 0 1px no-repeat;
 }
 
 .icon-forum {
     width: 22px;
     padding-left: 22px;
-    background: url(../../../images/forum.png) 0 1px no-repeat;
+    background: url(../../images/forum.png) 0 1px no-repeat;
 }
 
 .PhorumAdminTableHead
@@ -154,7 +154,7 @@
 
 .PhorumAdminError
 {
-    background-image: url("../../../images/alert.gif");
+    background-image: url("../../images/alert.gif");
     background-position: 5px 5px;
     background-repeat: no-repeat;
     font-family: Lucida Grande, Lucida Sans Unicode, Verdana, Arial, Helvetica;
--- phorum-5.2.14/include/admin/cache.php~	2010-01-11 16:30:31.000000000 +0200
+++ phorum-5.2.14/include/admin/cache.php	2010-01-11 16:58:45.262528168 +0200
@@ -32,7 +32,7 @@
             case "cache":
 
                 if ( empty( $value ) ) {
-                    $new_settings[$field] = "/tmp";
+                    $new_settings[$field] = "/var/cache/phorum";
                 } elseif ( !file_exists( $value ) ) {
                     $error = "This cache directory does not exist.  Please create it with the proper permissions.";
                 } else {
--- phorum/include/db/mysql.php~	2010-01-12 18:38:55.000000000 +0200
+++ phorum/include/db/mysql.php	2010-01-12 18:42:08.086600820 +0200
@@ -7636,7 +7636,7 @@
 );
 
 // Load the specific code for the PHP extension that we use.
-$extfile = "./include/db/mysql/{$ext}.php";
+$extfile = PHORUM_INCLUDES_DIR."/db/mysql/{$ext}.php";
 if (!file_exists($extfile)) trigger_error(
    "The Phorum MySQL database layer is unable to find the extension " .
    "file $extfile on the system. Check if all Phorum files are uploaded " .
--- phorum-5.2.14/htdocs/ajax.php~	2010-01-12 19:06:49.000000000 +0200
+++ phorum-5.2.14/htdocs/ajax.php	2010-01-12 19:07:09.826450999 +0200
@@ -154,7 +154,7 @@
 }
 
 // Check if the Ajax call has a core handler script.
-if (file_exists("./include/ajax/call.$ajax_call.php")) {
+if (file_exists(PHORUM_INCLUDES_DIR."/ajax/call.$ajax_call.php")) {
     include PHORUM_INCLUDES_DIR.'/ajax/call.$ajax_call.php';
     exit();
 }
--- phorum/htdocs/javascript.php	2010-01-12 19:00:24.036492037 +0200
+++ phorum/htdocs/javascript.php	2010-01-12 19:02:34.413145270 +0200
@@ -31,15 +31,15 @@
 $module_registrations[] = array(
     'module'    => 'core',
     'source'    => 'file(include/ajax/client.js.php)',
-    'cache_key' => filemtime('./include/ajax/client.js.php') .
+    'cache_key' => filemtime(PHORUM_INCLUDES_DIR.'/ajax/client.js.php') .
                    $PHORUM['DATA']['URL']['AJAX']
 );
 
 // Add template specific javascript code, if available. The template writer
 // can put the javascript code to include in the file
 // "templates/<name>/javascript.tpl" or "templates/<name>/javascript.php".
-if (file_exists("./templates/{$PHORUM['template']}/javascript.tpl") ||
-    file_exists("./templates/{$PHORUM['template']}/javascript.php")) {
+if (file_exists(PHORUM_DIR."/templates/{$PHORUM['template']}/javascript.tpl") ||
+    file_exists(PHORUM_DIR."/templates/{$PHORUM['template']}/javascript.php")) {
     $module_registrations[] = array(
         'module' => $PHORUM['template'] . ' template',
         'source' => 'template(javascript)'
@@ -234,10 +234,10 @@
         {
             case "file":
                 ob_start();
-                // Make sure that relative paths start with "./",
-                // just in case "." is not in the PHP include path.
+                // Make relative paths start with PHORUM_DIR,
+                // as we moved files around.
                 $path = $r['source'][0] == '/'
-                      ? $r['source'] : './'.$r['source'];
+                      ? $r['source'] : PHORUM_DIR.'/'.$r['source'];
                 include($path);
                 $content .= ob_get_contents();
                 ob_end_clean();
--- phorum/htdocs/css.php~	2010-01-12 19:27:06.000000000 +0200
+++ phorum/htdocs/css.php	2010-01-12 19:30:47.253394705 +0200
@@ -274,7 +274,11 @@
         {
             case "file":
                 ob_start();
-                include($r['source']);
+                // Make relative paths start with PHORUM_DIR,
+                // as we moved files around.
+                $path = $r['source'][0] == '/'
+                     ? $r['source'] : PHORUM_DIR.'/'.$r['source'];
+                include($path);
                 $add .= ob_get_contents();
                 ob_end_clean();
                 break;
--- phorum/include/admin/header.php~	2010-01-13 19:03:16.000000000 +0200
+++ phorum/include/admin/header.php	2010-01-13 21:30:22.607156616 +0200
@@ -25,7 +25,7 @@
         if (!file_exists("./include/lang/${lang}.php")) {
             $lang = PHORUM_DEFAULT_LANGUAGE;
         }
-        require_once PHORUM_INCLUDES_DIR.'/lang/{$lang}.php';
+        require_once PHORUM_INCLUDES_DIR."/lang/{$lang}.php";
     }
 
     // HTTP Content-Type header with the charset from the default language
--- phorum/htdocs/admin.php	2010-01-13 21:31:15.510488551 +0200
+++ phorum/htdocs/admin.php	2010-01-13 21:33:38.367557450 +0200
@@ -77,7 +77,7 @@
             } elseif(isset($_GET["module"]) && is_scalar($_GET["module"])){
                 $module = @basename($_GET["module"]);
             }
-            if(empty($module) || !file_exists("./include/admin/$module.php")){
+            if(empty($module) || !file_exists(PHORUM_INCLUDES_DIR."/admin/$module.php")){
                 $module = "default";
             }
             // check the admin token
@@ -105,7 +105,7 @@
     $module = phorum_hook( "admin_pre", $module );
     ob_start();
     if($module!="help") require_once PHORUM_INCLUDES_DIR.'/admin/header.php';
-    require_once PHORUM_INCLUDES_DIR.'/admin/$module.php';
+    require_once PHORUM_INCLUDES_DIR."/admin/$module.php";
     if($module!="help") require_once PHORUM_INCLUDES_DIR.'/admin/footer.php';
     ob_end_flush();
 
--- phorum/include/api/modules.php	2010-01-13 21:36:05.906916979 +0200
+++ phorum-5.2.14/include/api/modules.php	2010-01-13 22:41:33.709858271 +0200
@@ -95,10 +95,10 @@
 
     require_once PHORUM_INCLUDES_DIR.'/version_functions.php';
 
-    $dh = opendir("./mods");
+    $dh = opendir(PHORUM_DIR."/mods");
     if (! $dh) trigger_error(
         "Unable to create a list of available modules: " .
-        "opendir of directory \"./mods\" failed.",
+        "opendir of directory \"mods\" failed.",
         E_USER_ERROR
     );
 
@@ -116,11 +116,11 @@
 
         // Read in the module information.
         $lines = array();
-        if (file_exists("./mods/$entry/info.txt")) {
-            $lines = file("./mods/$entry/info.txt");
-        } elseif (is_file("./mods/$entry") && substr($entry, -4)==".php") {
+        if (file_exists(PHORUM_DIR."/mods/$entry/info.txt")) {
+            $lines = file(PHORUM_DIR."/mods/$entry/info.txt");
+        } elseif (is_file(PHORUM_DIR."/mods/$entry") && substr($entry, -4)==".php") {
             $entry = str_replace(".php", "", $entry);
-            $data = file_get_contents("./mods/$entry.php");
+            $data = file_get_contents(PHORUM_DIR."/mods/$entry.php");
             if($data = stristr($data, "/* phorum module info")){
                 $data = substr($data, 0, strpos($data, "*/"));
                 $lines = preg_split('!(\r|\n|\r\n)!', $data);
@@ -196,7 +196,7 @@
             $info['enabled'] = 0;
         }
 
-        if (file_exists("./mods/$entry/settings.php")){
+        if (file_exists(PHORUM_DIR."/mods/$entry/settings.php")){
             $info["settings"]=true;
         } else {
             $info["settings"]=false;
@@ -524,8 +524,8 @@
         foreach ($PHORUM['mods'] as $mod => $active)
         {
             if (!$active) continue;
-            $info = "./mods/$mod/info.txt";
-            $filemod = "./mods/$mod.php";
+            $info = PHORUM_DIR."/mods/$mod/info.txt";
+            $filemod = PHORUM_DIR."/mods/$mod.php";
             if (file_exists($info)) {
                 $time = @filemtime($info);
             } elseif (file_exists($filemod)) {
--- phorum-5.2.14/include/admin/manage_languages.php~	2010-01-13 22:36:06.000000000 +0200
+++ phorum-5.2.14/include/admin/manage_languages.php	2010-01-13 22:39:37.387371140 +0200
@@ -665,7 +665,7 @@
 
     // For the announcement module, we keep the language strings in
     // the main langage file.
-    phorum_extract_language_strings_recurse("./mods/announcements");
+    phorum_extract_language_strings_recurse(PHORUM_DIR."/mods/announcements");
 
     return $extract_strings;
 }
--- phorum-5.2.14/include/admin/mods.php~	2010-01-13 22:36:06.000000000 +0200
+++ phorum-5.2.14/include/admin/mods.php	2010-01-13 22:40:09.750492277 +0200
@@ -110,7 +110,7 @@
         $moreinfo[] = "<a href=\"".$info["url"]."\">web site</a>";
     }
     foreach(array('README','INSTALL','Changelog') as $file) {
-        if(file_exists("./mods/$name/$file")){
+        if(file_exists(PHORUM_DIR."/mods/$name/$file")){
             $moreinfo[] = "<a target=\"_blank\" href=\"".$PHORUM["http_path"]."/mods/$name/$file\">$file</a>";
         }
     }
--- phorum-5.2.14/include/admin/modsettings.php~	2010-01-13 22:36:06.000000000 +0200
+++ phorum-5.2.14/include/admin/modsettings.php	2010-01-13 22:40:38.894030607 +0200
@@ -21,12 +21,12 @@
 
     $mod = basename($_REQUEST["mod"]);
 
-    if(file_exists("./mods/$mod/settings.php")){
+    if(file_exists(PHORUM_DIR."/mods/$mod/settings.php")){
         if (!isset($PHORUM["mods"][$mod]) || !$PHORUM["mods"][$mod]) {
             $text = "This module is not enabled yet. You can change its settings but the module is only active if you enable it on the previous page.";
             phorum_admin_error($text);
         }
-        include_once("./mods/$mod/settings.php");
+        include_once(PHORUM_DIR."/mods/$mod/settings.php");
 
     } else {
 
--- phorum-5.2.14/include/admin/sanity_checks/modules_filenames.php~	2010-01-13 22:36:06.000000000 +0200
+++ phorum-5.2.14/include/admin/sanity_checks/modules_filenames.php	2010-01-13 22:41:12.744027348 +0200
@@ -29,7 +29,7 @@
         }
 
 
-        $d = dir("./mods");
+        $d = dir(PHORUM_DIR."/mods");
         while (false !== ($entry = $d->read()))
         {
         	// Some entries which we skip by default.
@@ -39,20 +39,20 @@
         	
         	// Read in the module information.
         	$lines = array();
-        	if (file_exists("./mods/$entry/info.txt")) {
-        		$lines = file("./mods/$entry/info.txt");
+        	if (file_exists(PHORUM_DIR."/mods/$entry/info.txt")) {
+        		$lines = file(PHORUM_DIR."/mods/$entry/info.txt");
         	} 
         	
-        	if(is_file("./mods/$entry") && substr($entry, -4)==".php") {
+        	if(is_file(PHORUM_DIR."/mods/$entry") && substr($entry, -4)==".php") {
         		// one file module, skip it
         	} else {
-        		if(!file_exists("./mods/$entry/info.txt")) {
+        		if(!file_exists(PHORUM_DIR."/mods/$entry/info.txt")) {
         			return array(
                     PHORUM_SANITY_WARN,
                     "Your module &quot;$entry&quot; doesn't have an info.txt file in its directory. Either its not a module or the installation of that module is broken.", 
                     "You should remove all files or directories which are not modules from the mods-directory and fix broken module installations."
                     );
-        		} elseif (!file_exists("./mods/$entry/$entry.php")) {
+        		} elseif (!file_exists(PHORUM_DIR."/mods/$entry/$entry.php")) {
         			return array(
                     PHORUM_SANITY_WARN,
                     "Your module &quot;$entry&quot; doesn't have an corresponding .php file in its directory. 
--- phorum-5.2.14/mods/username_restrictions/settings.php~	2010-01-13 22:36:06.000000000 +0200
+++ phorum-5.2.14/mods/username_restrictions/settings.php	2010-01-13 22:41:43.609857184 +0200
@@ -1,7 +1,7 @@
 <?php
     if (!defined("PHORUM_ADMIN")) return;
 
-    include("./mods/username_restrictions/defaults.php");
+    include(PHORUM_DIR."/mods/username_restrictions/defaults.php");
 
     // Options for the valid username characters.
     $valid_chars_options = array(
--- phorum-5.2.14/mods/bbcode/bbcode.php	2010-01-13 22:43:01.370682640 +0200
+++ phorum-5.2.14/mods/bbcode/bbcode.php	2010-01-13 22:48:35.809816117 +0200
@@ -210,7 +210,7 @@
         editor_tools_register_tool(
             $id,                           // Tool id
             $description,                  // Tool description
-            "./mods/bbcode/icons/$id.gif", // Tool button icon
+            "mods/bbcode/icons/$id.gif", // Tool button icon
             "editor_tools_handle_$id()"    // Javascript action on click
         );
     }
@@ -285,10 +285,10 @@
     if ($PHORUM["args"]["action"] == 'help')
     {
         $lang = $GLOBALS['PHORUM']['language'];
-        if (!file_exists("./mods/bbcode/help/$lang/bbcode.php")) {
+        if (!file_exists(PHORUM_DIR."/mods/bbcode/help/$lang/bbcode.php")) {
             $lang = 'english';
         }
-        include("./mods/bbcode/help/$lang/bbcode.php");
+        include(PHORUM_DIR."/mods/bbcode/help/$lang/bbcode.php");
         exit(0);
     }
 
--- phorum-5.2.14/mods/event_logging/db.php~	2010-01-13 22:36:06.000000000 +0200
+++ phorum-5.2.14/mods/event_logging/db.php	2010-01-13 22:43:26.263180363 +0200
@@ -30,7 +30,7 @@
         $version++;
         $settings = array( "mod_event_logging_installed" => $version );
 
-        $sqlfile = "./mods/event_logging/db/" .
+        $sqlfile = PHORUM_DIR."/mods/event_logging/db/" .
                    $PHORUM["DBCONFIG"]["type"] . "/$version.php";
 
         if (! file_exists($sqlfile)) {
--- phorum-5.2.14/mods/event_logging/settings.php~	2010-01-13 22:36:06.000000000 +0200
+++ phorum-5.2.14/mods/event_logging/settings.php	2010-01-13 22:43:46.897345026 +0200
@@ -49,7 +49,7 @@
 
 // Load the settings screen. For security,
 // we follow a strict naming scheme here.
-$settings_file = "./mods/event_logging/settings/{$action}.php";
+$settings_file = PHORUM_DIR."/mods/event_logging/settings/{$action}.php";
 if (file_exists($settings_file)) {
     include($settings_file);
 } else {
--- phorum-5.2.14/mods/smileys/help/english/smileys.php~	2010-01-13 22:36:06.000000000 +0200
+++ phorum-5.2.14/mods/smileys/help/english/smileys.php	2010-01-13 22:44:06.914009661 +0200
@@ -29,7 +29,7 @@
       <th class="PhorumAdminTableHead">Description</th>
       <th class="PhorumAdminTableHead">Where</th>
     </tr>
-    <?php include("./mods/smileys/help/render_smileys_list.php") ?>
+    <?php include(PHORUM_DIR."/mods/smileys/help/render_smileys_list.php") ?>
     </table>
 
     <br/><br/><br/>
--- phorum-5.2.14/mods/smileys/smileys.php~	2010-01-13 22:36:06.000000000 +0200
+++ phorum-5.2.14/mods/smileys/smileys.php	2010-01-13 22:44:30.416506930 +0200
@@ -95,7 +95,7 @@
         editor_tools_register_tool(
             'smiley',                              // Tool id
             $lang['smiley'],                       // Tool description
-            "./mods/smileys/icon.gif",             // Tool button icon
+            "mods/smileys/icon.gif",             // Tool button icon
             "editor_tools_handle_smiley()",        // Javascript click action
             NULL,                                  // Tool icon width
             NULL,                                  // Tool icon height
@@ -109,7 +109,7 @@
         editor_tools_register_tool(
             'subjectsmiley',                       // Tool id
             $lang['subjectsmiley'],                // Tool description
-            "./mods/smileys/icon.gif",             // Tool button icon
+            "mods/smileys/icon.gif",             // Tool button icon
             "editor_tools_handle_subjectsmiley()", // Javascript click action
             NULL,                                  // Tool icon width
             NULL,                                  // Tool icon height
@@ -141,10 +141,10 @@
     if ($PHORUM["args"]["action"] == 'help')
     {
         $lang = $GLOBALS['PHORUM']['language'];
-        if (!file_exists("./mods/smileys/help/$lang/smileys.php")) {
+        if (!file_exists(PHORUM_DIR."/mods/smileys/help/$lang/smileys.php")) {
             $lang = 'english';
         }
-        include("./mods/smileys/help/$lang/smileys.php");
+        include(PHORUM_DIR."/mods/smileys/help/$lang/smileys.php");
         exit(0);
     }
 
--- phorum-5.2.14/mods/smileys/smileyslib.php~	2010-01-13 22:36:06.000000000 +0200
+++ phorum-5.2.14/mods/smileys/smileyslib.php	2010-01-13 22:44:58.696505207 +0200
@@ -18,7 +18,7 @@
 
 // The default smiley prefix path.
 global $MOD_SMILEY_DEFAULT_PREFIX;
-$MOD_SMILEY_DEFAULT_PREFIX = './mods/smileys/images/';
+$MOD_SMILEY_DEFAULT_PREFIX = 'mods/smileys/images/';
 
 // The default list of smileys to install upon initial setup.
 global $MOD_SMILEY_DEFAULT_SMILEYS;
--- phorum-5.2.14/mods/spamhurdles/db.php~	2010-01-13 22:36:06.000000000 +0200
+++ phorum-5.2.14/mods/spamhurdles/db.php	2010-01-13 22:45:21.459835060 +0200
@@ -17,7 +17,7 @@
 {
     global $PHORUM;
 
-    $layerpath = "./mods/spamhurdles/db/{$PHORUM["DBCONFIG"]["type"]}";
+    $layerpath = PHORUM_DIR."/mods/spamhurdles/db/{$PHORUM["DBCONFIG"]["type"]}";
 
     // Allow db layers to provide an initialization script of their own.
     // The main goal for this script is to allow a db layer to override the
--- phorum-5.2.14/mods/spamhurdles/spamhurdles.php~	2010-01-13 22:36:06.000000000 +0200
+++ phorum-5.2.14/mods/spamhurdles/spamhurdles.php	2010-01-13 22:45:36.219833260 +0200
@@ -92,7 +92,7 @@
     // If a spoken captcha is requested, then generate a wav file for it
     // and stream it to the user.
     if (isset($PHORUM["args"]["spokencaptcha"])) {
-        include("./mods/spamhurdles/captcha/spoken_captcha.php");
+        include(PHORUM_DIR."/mods/spamhurdles/captcha/spoken_captcha.php");
         exit(0);
     }
 
--- phorum/include/admin/forum_defaults.php~	2010-01-13 23:12:00.000000000 +0200
+++ phorum/include/admin/forum_defaults.php	2010-01-13 23:16:15.052984229 +0200
@@ -21,6 +21,6 @@
 
     define("PHORUM_DEFAULT_OPTIONS", 1);
 
-    include "./include/admin/newforum.php";
+    require_once PHORUM_INCLUDES_DIR.'/admin/newforum.php';
 
 ?>
--- phorum/include/admin/editforum.php~	2010-01-14 12:34:47.000000000 +0200
+++ phorum/include/admin/editforum.php	2010-01-14 12:34:48.878192205 +0200
@@ -24,7 +24,7 @@
     if(empty($_REQUEST["forum_id"])){
         phorum_admin_error("forum_id not set");
     } else {
-        include "./include/admin/newforum.php";
+        include PHORUM_INCLUDES_DIR."/admin/newforum.php";
     }
 
 ?>
--- phorum/include/admin/sanity_checks.php~	2010-02-10 17:19:10.000000000 +0200
+++ phorum/include/admin/sanity_checks.php	2010-02-10 17:33:03.766530721 +0200
@@ -20,7 +20,7 @@
 if(!defined("PHORUM")) return;
 
 // The place where our sanity checking modules are.
-$sanity_checks_dir = "./include/admin/sanity_checks";
+$sanity_checks_dir = PHORUM_INCLUDES_DIR."/admin/sanity_checks";
 
 // Mapping of status to display representation.
 $status2display = array(
