--- phorum/mods/spamhurdles/captcha/class.captcha_recaptcha.php~	2010-01-13 22:21:20.000000000 +0200
+++ phorum/mods/spamhurdles/captcha/class.captcha_recaptcha.php	2010-01-13 22:35:22.516978432 +0200
@@ -11,7 +11,7 @@
 // we might implement the configuration as a part of the CAPTCHA class.
 // For now, this setup will do just fine.
 
-define('RECAPTCHA_LIB', './mods/spamhurdles/captcha/recaptcha-php-1.9/recaptchalib.php');
+define('RECAPTCHA_LIB', '/usr/share/php/recaptchalib.php');
 
 class captcha_recaptcha
 {
