--- phorum-5.2.23/mods/smtp_mail/smtp_mail.php~	2020-11-24 12:26:36.000000000 +0200
+++ phorum-5.2.23/mods/smtp_mail/smtp_mail.php	2020-11-24 12:27:06.033097480 +0200
@@ -35,10 +35,9 @@
 
         try {
 
-            require_once PHORUM_DIR.'/mods/smtp_mail/phpmailer/class.phpmailer.php';
+            require_once 'class.phpmailer.php';
 
             $mail = new PHPMailer();
-            $mail->PluginDir = "./mods/smtp_mail/phpmailer/";
 
             $mail->CharSet  = $PHORUM["DATA"]["CHARSET"];
             $mail->Encoding = $PHORUM["DATA"]["MAILENCODING"];
